x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats_top_list.csv-spec

```diff
@@ -0,0 +1,100 @@
+topList
```

Comment:
You need to add some more complex queries here.
For one, I don't see any `stats top_list by field` queries. Have a look at `stats.csv-spec` and take inspiration from there. A quick list to include:
- `stats top_list by field`
- `stats top_list by field1, field2`
- `stats x=top_list | sort x asc/desc`
- `row values | stats ....`
- have top_list act on dynamically created values (not something coming from Lucene or from `row`). For example, `from employees | STATS a = TOP_VALUES_LIST(salary_change, 3, \"desc\"), d = TOP_VALUES_LIST(salary_change, 3, \"desc\") by gender | sort a desc, d asc | stats ad = top_values_list(a, 9, \"desc\"), da = top_values_list(d, 9, \"asc\") | eval dad = mv_sort(da, \"desc\") | keep ad, dad`

Reply:
Yeah, these are all good things. These tests are quite cheap to run so I'm fully behind writing a bunch of htem.

Reply:
Added some extra cases, like the ones you commented. I'm never sure about how much to test, as at some point it feels like testing the engine itself

Reply:
My suggestions come from the query languages side of things. In my view, your csv-spec tests are great but pretty basic. I am looking for combination of commands and functions, nested functions and repeating commands and so on. Something that can bring out issues with planning, analysis, optimization steps that might require some special attention from a new `stats` command. My advice going forward is to seek such more complex queries in already existent csv-spec tests, there are some good ones in there.